addons/sale_project/models/project.py

```diff
@@ -869,6 +869,11 @@ def _inverse_partner_id(self):
             if task.sale_order_id and task.partner_id.commercial_partner_id not in consistent_partners:
                 task.sale_order_id = task.sale_line_id = False
 
+    @api.onchange('partner_id')
```

Comment:
Hello, 
Good job,

IMO, i think we can do some fixes to make it clean and avoid adding this onchange

1- updating task.partner (line 844) doesn't make a sens inside _compute_sale_order_id, we should only update field x inside a _compute_x, so we should move this update inside _compute_partner_id and add sale_line_id, project_id and sale_order_id in the @api.depends decorator (i know it's not your code, but it's an opportunity to improve it)

2- in _inverse_partner_id, when sale_line_id, sale_order_id are computed, to avoid having a compute cycle, we can use self.env.add_to_compute() method to suppose that _compute_partner_id is already computed and it's not recomputed again

3- move your logic inside _inverse_partner_id

Let me know if something is not clear

Reply:
Hello @RaoufGhrissi ,

I have completely remove the onchange and did revert changes in the PR's mentioned in the PR Commit message.After I dont think we would even need _inverse function as everyting can be handled in compute functions themselves.

Those PR's were the culprits that were causing the bug :thinking: .

You can have a look and let me know what you think :smile:

Reply:
Hello @RaoufGhrissi I have gone through your suggestion the point is absolutely true as it may cause future regressions. About second point I aim to reverse what is done in the [PR](https://github.com/odoo/odoo/pull/132320) this is the PR from which we get the following bug so I would likely remove inverse if possible.Also as `partner_id.commercial_id` is removed from depends and after that many changes have been done I am getting multiple infinite computations, I am making sure if it is really possible to revert the changes in the PR above. I will let you know of any my finding soon. Sorry for the late message I was stuck another blocking task.

Reply:
Hello again @RaoufGhrissi , We may not be able to revert the PR as the implemented feature becomes obsolete :cry: . Here [this](https://github.com/odoo/odoo/blob/9e40880b1ae7295dbd49eb1bb3ae0be84a137590/addons/sale_project/models/project.py#L841) is causing a infinite loop and here is where we get sale_order_id to implement the feature in PR. There sale_line_id is a compute field and is a recursive field. We could use SQL to fetch the values but that is really unwarranted I think. What do you think ?

IMO, onchange here serves the purpose and is very simple and is not that extensive (just sets `sale_line_id` False). If you have any suggestions please do suggest me. I could may send you a patch/diff for you test locally with the changes I have made recently (The changes I haven't pushed).

Thank You.

Reply:
Hey @kthe-odoo  thanks for your work and investigation, yes i know, it's simple to do this task with an onchange but i'm afraid that this code become too complicated, we have an onchange, inverse and a compute doing 2 writes, have you tried stoping the infiite loop using `self.env.add_to_compute()` ? it's used when we say that it was computed so no need to call the compute again ? if you want, you can push your changes instead of sending the diff, i can pull and test it locally, maybe i can help you. 

Thank you

Reply:
Hello @RaoufGhrissi , The infinite computation occours only with tasks with timesheets linked to task, this is due to the fact timesheets change sale_line_id when task sale_line_id is changed(because of depend). I have tried add_to_compute and but don't you think all fields in the `self.all.tocompute` get executes after compute function is done or onchange is done. Am i wrong? I have pushed the code and left some inline comments for you.

Have a nice weekend.

Reply:
hahaha too much details for Friday afternoon 😵‍💫😵‍💫😵‍💫😵‍💫 but let's see if we can improve it.

Have a nice weekend

Reply:
hey @kthe-odoo  i took a look to your code and i have a question, in my first comment (point 2 and 3), i was suggesting keeping _inverse_partner_id as the inverse logic already protect from loops (https://www.odoo.com/documentation/18.0/developer/tutorials/server_framework_101/08_compute_onchange.html#inverse-function) but in your code, you deleted it. what was the reason, i'm asking to avoid reinventing something that you have tested

Reply:
hey @kthe-odoo , i took a look and yes 😵‍💫 it's confusing, we have compute_so computing partner also, compute_partner, inverse_partner..., i was trying some changes but i wasn't confident about what i'm doing, so as they task is high priority, let's advance with your onchange logic, maybe this code can be refactored in another bigger task. 

sorry for wasting your time but i think it's worth it, we discovered some new code

Reply:
Hey @RaoufGhrissi , Now you understand what I have been going through :smiling_face_with_tear:. I can assure you that my time didnt go to waste as I have learn about odoo python framework, brushed up my python skills as well. I would have gotten a chance to dive deep if you hadn't asked me to. I am reverting to the onchange solution.

![gifgit (4)-min](https://github.com/user-attachments/assets/c1fe35e2-7407-479c-b657-0c6fdc868d6f)